src/pages/gutenberg/free-audio-books.vue

```diff
@@ -291,6 +359,52 @@ export default {
     scrollToTop() {
       window.scrollTo({ top: 0, behavior: 'smooth' });
     },
+    handleLoadMore() {
+      this.currentDisplayNumber += 100;
+    },
+    handleInputChange(value) {
+      logTrackerEvent(this, 'Gutenberg', 'inputChange', value.target.value, 1);
+    },
   },
 };
+
+function parseCSV(csvData) {
+  return new Promise((resolve, reject) => {
+    const parsedData = [];
+    const csvStream = csvParser();
+
+    csvStream.on('data', record => {
+      parsedData.push(record);
+    });
+
+    csvStream.on('end', () => {
+      resolve(parsedData);
+    });
+
+    csvStream.on('error', err => {
+      reject(err);
+    });
+
+    csvStream.write(csvData);
+    csvStream.end();
+  });
+}
+
+function generateJSONLD(data) {
+  if (!Array.isArray(data)) {
+    return {};
+  }
+
+  return {
+    '@context': 'https://schema.org',
+    '@type': 'ItemList',
+    itemListElement: data.map((item, index) => ({
+      '@type': 'ListItem',
+      position: index + 1,
+      name: item.classTitle,
+      author: item.author,
+      classId: item.classId,
+    })),
```

Comment:
Please use `DataList` of `Book` , like in `/store/featured` etc pages
https://developers.google.com/search/docs/appearance/structured-data/book#datafeed-entity

Reply:
[♻️ Simplify generate book JSONLD](https://github.com/likecoin/liker-land/pull/1620/commits/45cb67c442d0ca66439eb2b43e5327fe35b5423f)